src/protozero/filtering/filter_bytecode_parser.cc

```diff
@@ -170,6 +337,7 @@ bool FilterBytecodeParser::LoadInternal(const uint8_t* bytecode_data,
       words_.insert(words_.end(), ranges.begin(), ranges.end());
       direct_indexed_fields.clear();
       ranges.clear();
+      ++current_msg_index;
```

Comment:
can we add some sanitization step at the end that checks that everything is in order (e.g. ranges don't overlap)